libcaf_test/caf/test/fixture/deterministic.cpp

```diff
@@ -514,6 +502,32 @@ detail::mailbox_factory* deterministic::config::mailbox_factory() {
   return factory_.get();
 }
 
+// -- system -------------------------------------------------------------------
+
+deterministic::system_impl::system_impl(actor_system_config& cfg,
+                                        deterministic* fix)
+  : actor_system(cfg, custom_setup, fix) {
+  // nop
+}
+
+detail::actor_local_printer_ptr
+deterministic::system_impl::printer_for(local_actor* self) {
+  auto& ptr = printers_[self->id()];
+  if (!ptr)
+    ptr = make_counted<actor_local_printer_impl>(self);
+  return ptr;
+}
+
+void deterministic::system_impl::custom_setup(actor_system& sys,
+                                              actor_system_config& cfg,
+                                              void* custom_setup_data) {
+  auto* fix = static_cast<deterministic*>(custom_setup_data);
```

Comment:
Bit curious why you needed to cast `void*` here? My guess is you didn't want to expose deterministic to `caf_core`?

Reply:
It's just the most generic interface that allows us to hook basically any initialization code into the system. Right now, we just have the deterministic fixture here, but I'm certain we'll have other use cases at some point.